dripline/extensions/thermo_fisher_endpoint.py

```diff
@@ -0,0 +1,82 @@
+from dripline.core import Entity, calibrate, ThrowReply
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = []
+__all__.append("ThermoFisherGetHexEntity")
+__all__.append("ThermoFisherGetEntity")
+
+
+class ThermoFisherGetHexEntity(Entity):
+    '''
+    A simple endpoint which stores a value but returns that value + a random offset
+    '''
+
+    def __init__(self, 
+                 get_str=None,
+                 **kwargs):
+        '''
+        Args:
+            cmd_str: hexstring of the command, e.g. 20 
+        '''
+        if get_str is None:
+            raise ValueError('<base_str is required to __init__ ThermoFisherGetEntity instance')
+        else:
+            self.cmd_str = str(get_str).zfill(2)
+        Entity.__init__(self, **kwargs)
+
+    @calibrate()
+    def on_get(self):
+        # setup cmd here
+        to_send = [self.cmd_str]
+        logger.debug(f'Send cmd in hexstr: {to_send[0]}')
+        result = self.service.send_to_device(to_send)
+        logger.debug(f'raw result is: {result}')
+
+        return result
+
+    def on_set(self, value):       
+        raise ThrowReply('message_error_invalid_method', f"endpoint '{self.name}' does not support set")
+        to_send = [cmd]
+        result = self.service.send_to_device(to_send)
+        logger.debug(f'raw result is: {result}')
+        # do something here
+        return result
+
+
+class ThermoFisherGetEntity(ThermoFisherGetHexEntity):
+
+    units = {0: "",
+             1: "degC",
+             2: "degF",
+             3: "L/min",
+             4: "gal/min",
+             5: "sec",
+             6: "PSI",
+             7: "bar",
+             8: "MOhm cm",
+             9: "%",
+             10: "V",
+             11: "kPa",
+            }
+
+    def __init__(self, **kwargs):
+        ThermoFisherGetHexEntity.__init__(self, **kwargs)
+
+    @calibrate()
+    def on_get(self):
+        # setup cmd here
+        to_send = [self.cmd_str]
+        logger.debug(f'Send cmd in hexstr: {to_send[0]}')
+        result = self.service.send_to_device(to_send)
+        logger.debug(f'raw result is: {result}')
+ 
+        # do something here
+        decimal = 10.**(-int(result[0], 16))
+        unit = self.units[int(result[1], 16)]
```

Comment:
for now unit isn't used?

Reply:
Unit is constant and does not change. We either can return f"{value} {unit}" as a string or we just return the value as numeric value. If I am right we can not have a tuple as payload. If its a string that contains the units as well, then you would need a calibration function that strips the unit again and makes it a float. I see pros and cons for both versions. As you may see I evaluate both, but just return the value. I had a version before where I had value and unit as a string.